### ISSUE TITLE
Inlay hints colors are inconsistent

### ISSUE THREAD

**Issue by lucario387** [neovim/neovim#24068]
### Problem

Like the title described, the color of the inlay hints are inconsistent, depending on the color of the text before it

Reproducible with this source code
```c
#include <stdio.h>

int main() {
  printf("Hello World");
  printf( "Hello World"); // The color of `format` inlay hints is different from above
  return 0;
}
```

### Steps to reproduce using "nvim -u minimal_init.lua"

--- CHANGE THESE
local pattern = 'c'
local cmd = {'clangd'}
-- Add files/folders here that indicate the root of a project
local root_markers = {'.git', '.editorconfig'}
-- Change to table with settings if required
local settings = vim.empty_dict()

vim.api.nvim_create_autocmd('FileType', {
  pattern = pattern,
  callback = function(args)
    local match = vim.fs.find(root_markers, { path = args.file, upward = true })[1]
    local root_dir = match and vim.fn.fnamemodify(match, ':p:h') or nil
    vim.lsp.start({
      name = 'clangd',
      cmd = cmd,
      root_dir = root_dir,
      settings = settings
    })
  end
})

### Expected behavior

Both format to have the same color/hlgroup.

This is also reproducible with jdtls v1.24.0, and possibly other servers as well

### Neovim version (nvim -v)

NVIM v0.10.0-dev-540+gca5de9306

### Language server name/version

clangd version 15.0.7

### Operating system/version

Arch Linux

**Comment by lucario387** [comment]
Updated minimal_init.lua and title

**Comment by p00f** [comment]
Yeah I don't know why that happens, but defining `LspInlayHint` fixes it - should we define the highlight group in `highlight_group.c`?

**Comment by clason** [comment]
You can link it to `Normal` (or `Comment`?).

**Comment by justinmk** [comment]
Linking to `Special` (or some variant thereof) is probably most expected, similar to virtual text

**Comment by p00f** [comment]
@lucario387 #24073

**Comment by lucario387** [comment]
@p00f I can confirm that adding `LspInlayHint` hlgroup fixes the issue
